client/me/form-base/with-form-base.jsx

```diff
@@ -106,15 +64,28 @@ const withFormBase = ( WrappedComponent ) => {
 			toggleSetting: this.toggleSetting,
 			updateSetting: this.updateSetting,
 			submitForm: this.submitForm,
-			formState: this.state,
+			hasUnsavedUserSettings: this.props.hasUnsavedUserSettings,
 		} );
 
 		render() {
-			return <WrappedComponent { ...this.props } { ...this.getFormBaseProps() } />;
+			return (
+				<>
+					<QueryUserSettings />
+					<WrappedComponent { ...this.props } { ...this.getFormBaseProps() } />
+				</>
+			);
 		}
 	}
 
-	return connect( null, { errorNotice, successNotice } )( localize( EnhancedComponent ) );
+	return connect(
+		( state ) => ( {
+			_userSettings: getUserSettings( state ),
+			_unsavedUserSettings: getUnsavedUserSettings( state ),
+			hasUnsavedUserSettings: hasUnsavedUserSettings( state ),
+			isUpdatingUserSettings: isUpdatingUserSettings( state ),
```

Comment:
Pardon my potential ignorance in case I'm missing this, but this seems to be introducing pretty user-settings-specific functionality to this generic form base component. Is this the wrong place to be adding this functionality?  Or is this a temporary compromise in light of https://github.com/Automattic/wp-calypso/pull/48730/files#r554964900

Reply:
The form base mixin as well as the HOC are pretty tightly coupled with user settings, it's not a very general component (that's also a reason why I want to get rid of it). As we'll get rid of it eventually, yes to the latter.